code/modules/admin/admin_verbs.dm

```diff
@@ -690,21 +690,24 @@ var/list/admin_verbs_mod = list(
 	message_admins("<span class='notice'>[key_name_admin(usr)] gave [key_name(T)] the spell [S].</span>", 1)
 
 
-/client/proc/make_invisible(mob/T as mob in mob_list) // -- Urist
+/client/proc/toggle_invisible(mob/T as mob in mob_list)
 	set category = "Fun"
-	set name = "Make invisible"
+	set name = "Toggle invisiblity"
 	set desc = "Make a mob completely invisible."
 
-	T.alphas.len++
-	T.alphas[T.alphas.len] = 1
-	T.density = 0
-	T.forced_density = 1
+	if (T.alphas["admin_invis"] == 0)
+		T.forced_density = 0
+		T.alphas -= "admin_invis"
+	else	
+		T.alphas["admin_invis"] = 0
+		T.density = 0
```

Comment:
Why isn't the change to density mirrored?

Reply:
Density updates itself in life.dm